src/features/sanctions/hooks/useChainalysisSanctioned.ts

```diff
@@ -0,0 +1,43 @@
+import { isAddress } from 'viem';
+import { useContractRead } from 'wagmi';
+
+import { useEvmAccount } from '../../wallet/hooks/evm';
+
+// https://go.chainalysis.com/chainalysis-oracle-docs.html
+const ORACLE_ABI = [
+  {
+    inputs: [
+      {
+        internalType: 'address',
+        name: 'addr',
+        type: 'address',
+      },
+    ],
+    name: 'isSanctioned',
+    outputs: [
+      {
+        internalType: 'bool',
+        name: '',
+        type: 'bool',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+] as const;
+const ORACLE_ADDRESS = '0x40C57923924B5c5c5455c48D93317139ADDaC8fb';
+
+export const useChainalysisSanctioned = () => {
+  const evmAddress = useEvmAccount().addresses[0]?.address;
```

Comment:
Please confirm with @avious00 that it's sufficient for this task to just check EVM and not Cosmos and/or Solana. Typically new features for the Warp UI should be done in a multi-protocol way

Reply:
Totally agree but I couldn't find any lists to use for Cosmos or Solana.

This approach blocks all functionality for the app if you have a sanctioned wallet connected which feels like the correct behaviour.